tests/test_macros.h

```diff
@@ -471,3 +471,211 @@ class SignalWatcher : public Object {
+				std::fabs(lhs.origin.y - rhs.value.origin.y) <
+				rhs.epsilon * (rhs.scale + std::max<real_t>(std::fabs(lhs.origin.y), std::fabs(rhs.value.origin.y))) &&
+				std::fabs(lhs.origin.z - rhs.value.origin.z) <
+				rhs.epsilon * (rhs.scale + std::max<real_t>(std::fabs(lhs.origin.z), std::fabs(rhs.value.origin.z)));
+	}
+
+	friend bool operator==(AABB lhs, const GDApprox &rhs) {
+		return std::fabs(lhs.position.x - rhs.value.position.x) <
+				rhs.epsilon * (rhs.scale + std::max<real_t>(std::fabs(lhs.position.x), std::fabs(rhs.value.position.x))) &&
+				std::fabs(lhs.position.y - rhs.value.position.y) <
+				rhs.epsilon * (rhs.scale + std::max<real_t>(std::fabs(lhs.position.y), std::fabs(rhs.value.position.y))) &&
+				std::fabs(lhs.position.z - rhs.value.position.z) <
+				rhs.epsilon * (rhs.scale + std::max<real_t>(std::fabs(lhs.position.z), std::fabs(rhs.value.position.z))) &&
+				std::fabs(lhs.size.x - rhs.value.size.x) <
+				rhs.epsilon * (rhs.scale + std::max<real_t>(std::fabs(lhs.size.x), std::fabs(rhs.value.size.x))) &&
+				std::fabs(lhs.size.y - rhs.value.size.y) <
+				rhs.epsilon * (rhs.scale + std::max<real_t>(std::fabs(lhs.size.y), std::fabs(rhs.value.size.y))) &&
+				std::fabs(lhs.size.z - rhs.value.size.z) <
+				rhs.epsilon * (rhs.scale + std::max<real_t>(std::fabs(lhs.size.z), std::fabs(rhs.value.size.z)));
+	}
+
+	friend bool operator==(const GDApprox &lhs, T rhs) { return operator==(rhs, lhs); }
+	friend bool operator!=(T lhs, const GDApprox &rhs) { return !operator==(lhs, rhs); }
+	friend bool operator!=(const GDApprox &lhs, T rhs) { return !operator==(rhs, lhs); }
+	friend bool operator<=(T lhs, const GDApprox &rhs) { return lhs < rhs.value || lhs == rhs; }
+	friend bool operator<=(const GDApprox &lhs, T rhs) { return lhs.value < rhs || lhs == rhs; }
+	friend bool operator>=(T lhs, const GDApprox &rhs) { return lhs > rhs.value || lhs == rhs; }
+	friend bool operator>=(const GDApprox &lhs, T rhs) { return lhs.value > rhs || lhs == rhs; }
+	friend bool operator<(T lhs, const GDApprox &rhs) { return lhs < rhs.value && lhs != rhs; }
+	friend bool operator<(const GDApprox &lhs, T rhs) { return lhs.value < rhs && lhs != rhs; }
+	friend bool operator>(T lhs, const GDApprox &rhs) { return lhs > rhs.value && lhs != rhs; }
+	friend bool operator>(const GDApprox &lhs, T rhs) { return lhs.value > rhs && lhs != rhs; }
+
+	real_t epsilon;
+	real_t scale;
+	T value;
+};
+
+// Define a macro alias for the GDApprox helper type to improve code readability and conformance.
+#define APPROX(cond) GDApprox(cond)
+
+template <typename T>
+struct doctest::StringMaker<GDApprox<T>> {
+	static doctest::String convert(const GDApprox<T> &p_val) {
+		// Use the existing StringMaker for the value type
+		doctest::String result = doctest::StringMaker<T>::convert(p_val.value);
+		result += " (approximately, with epsilon: ";
+		result += doctest::toString(p_val.epsilon);
+		result += ")";
```

Comment:
Use this to print the text in gray, like in `main.cpp`'s `print_help()` function:

```suggestion
		result += " \u001b[90m(approximately, with epsilon: ";
		result += doctest::toString(p_val.epsilon);
		result += ")\u001b[0m";
```

![image](https://github.com/user-attachments/assets/48d68d95-1ccb-43fa-8df3-1d5be35222f1)

Reply:
Thanks a lot, I was trying to use doctest::Color but with no success. Well, if it works :)